src/routers/course.ts

```diff
@@ -32,41 +26,52 @@ courseRouter.post('/', async (req, res) => {
 	try {
 		const userID = req.header('userID');
 		if (userID) {
-			const course = new Course({
-				owner: userID,
-				name: req.body.name,
-			});
-			const created = await create(course);
-			res.json(created);
+			const body = req.body;
+			const checkPost = ajv.compile(
+				schema.components.schemas.CourseCreate
+			);
+
+			if (checkPost(body)) throw Error('body invalid');
+
+			const course = new Course({ ownerId: userID, ...body });
```

Comment:
You can remove the course here, you will make a different call once Seiya's stuff is in